File: fixtures/scale/java/large/src/cryptobench/ciphers/tea/TeaKey.java
package cryptobench.ciphers.tea;

public class TeaKey {
    private final int[] words;

    public TeaKey(int k0, int k1, int k2, int k3) {
        this.words = new int[] { k0, k1, k2, k3 };
    }

    public int k(int index) {
        return words[index & 3];
    }

    public static TeaKey defaultKey() {
        return new TeaKey(0x0123_4567, 0x89AB_CDEF, 0xFEDC_BA98, 0x7654_3210);
    }
}

File: fixtures/scale/dart/large/lib/ciphers/tea/tea_key.dart
class TeaKey {
  final int k0, k1, k2, k3;

  TeaKey(this.k0, this.k1, this.k2, this.k3);

  int k(int index) {
    switch (index & 3) {
      case 0:
        return k0;
      case 1:
        return k1;
      case 2:
        return k2;
      default:
        return k3;
    }
  }

  factory TeaKey.defaultKey() => TeaKey(0x01234567, 0x1A2B3C4D, 0x1234568, 0x76543210);
}
